### DISCUSSION TITLE
Is it possible for me to use it to implement something like apcu

### DISCUSSION THREAD

**Discussion by AliceSync** [zephir-lang/zephir#2423] [General]
If ok, can you provide a simple example, I'm learning to use it to implement some memory sharing aspects

**Comment by niden** [comment]
You can call any `apcu*` methods from Zephir in your class. You can check the `Phalcon\Storage` namespace, the `Apcu` adapter in particular. That will give you some ideas on how to use it

https://github.com/phalcon/cphalcon`

**Comment by AliceSync** [comment]
> You can call any methods from Zephir in your class. You can check the namespace, the adapter in particular. That will give you some ideas on how to use it`apcu*``Phalcon\Storage``Apcu`
> 
> [https://github.com/phalcon/cphalcon`](https://github.com/phalcon/cphalcon%60)

I attempted to install from the link you sent, and tried out "Phalcon\Cache\Cache" but it's not what I wanted. I think you misunderstood. What I want to achieve is my own shared memory, just like how apcu allows applications to share data even after a process forks, and it doesn't affect other non-forked independent processes. The link you provided uses an existing PHP extension.

**Comment by niden** [comment]
To build something like APC you will need to build first the server side, i.e. the application that will accept requests and store data in some form (files, memory or other storage mechanism) and then the client library that will connect to your server to get and store data.

Honestly this is more suited to be developed with C or some other language that is closer to the OS than Zephir is. It will all depend on your implementation and how you want to store that data and how to retrieve it.
